packages/cli/generation/ir-generator/src/converters/services/convertHttpService.ts

```diff
@@ -371,19 +377,24 @@ export function getHeaderName({ headerKey, header }: { headerKey: string; header
     };
 }
 
-function convertTransportToEncoding(transport: Transport): Encoding {
-    switch (transport.type) {
-        case "http":
-            return {
-                json: {},
-                proto: undefined
-            };
-        case "grpc":
-            return {
-                json: undefined,
-                proto: {}
-            };
-        default:
-            assertNever(transport);
+function convertEndpointSourceToEncoding(source: RawSchemas.SourceSchema | undefined): Encoding {
+    if (!source) {
+        return {
+            json: {},
+            proto: undefined
+        };
+    }
+    if ("openapi" in source) {
+        return {
+            json: {},
+            proto: undefined
+        };
+    }
+    if ("proto" in source) {
+        return {
+            json: undefined,
+            proto: {}
+        };
```

Comment:
We can use [isRawProtobufSourceSchema](https://github.com/fern-api/fern/blob/23bece6128e97891bd6ccb2954420732bdd8d0ba/packages/cli/fern-definition/schema/src/utils/isRawProtobufSourceSchema.ts#L3) for this - we could also roll out an `openapi` variant so that this reads a little more clearly.

Reply:
This has changed with the commit I didn't push.
It's now converting it based on the calculated transport and whether there are any HTTP endpoints mixed in.